src/Facets/StargateFacet.sol

```diff
@@ -184,28 +187,46 @@ contract StargateFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
     function _startBridge(
         ILiFi.BridgeData memory _bridgeData,
         StargateData calldata _stargateData
-    ) private noNativeAsset(_bridgeData) {
-        LibAsset.maxApproveERC20(
-            IERC20(_bridgeData.sendingAssetId),
-            address(router),
-            _bridgeData.minAmount
-        );
+    ) private {
+        if (LibAsset.isNativeAsset(_bridgeData.sendingAssetId)) {
+            nativeRouter.swapETHAndCall{ value: _bridgeData.minAmount }(
```

Comment:
Not sure but when bridge native asset, Stargate doesn't ask `lzFee`?

Reply:
From what I understand in order to behave like other bridges we would like to deduct the fee from the minAmount. Correct me if I'm wrong @maxklenk?

Reply:
maybe we can add `payFeeWithSendingAsset` to the `StargateData`.
and when the flag is set, then we include the fee in `minAmount`, otherwise, we separate the fee from `minAmount`

Reply:
The `payFeeWithSendingAsset` flag would make it clearer. But as we can't really use it for other tokens, because stargate does not allow to pay the fee with them, it could be confusing.

I see the beauty of using the including the fee into the transferred amount, as e.g. it would be possible to start with some ERC20 only, swap all of it into native, pay the fees with it and bridge the rest.

I have to check in the backend how the fees are actually calculated, I assume it works the same as with ERC20 tokens, that we specify the amount we want to transfer and the fees are added. This would make it difficult to start with an exact amount that includes both fee and transfer amount.